server/src/utils.ts

```diff
@@ -104,16 +104,32 @@ export let formatUsingValidBscNativePath = (
     encoding: "utf-8",
   });
   try {
-    let result = childProcess.execFileSync(bscNativePath, [
-      "-color",
-      "never",
-      "-format",
-      formatTempFileFullPath,
-    ]);
-    return {
-      kind: "success",
-      result: result.toString(),
-    };
+    // Default to using the project formatter. If not, use the one we ship with
+    // the analysis binary in the extension itself.
+    if (bscNativePath != null) {
+      let result = childProcess.execFileSync(bscNativePath, [
+        "-color",
+        "never",
+        "-format",
+        formatTempFileFullPath,
+      ]);
+      return {
+        kind: "success",
+        result: result.toString(),
+      };
+    } else {
+      let result = runAnalysisAfterSanityCheck(
```

Comment:
This is kind of a strange code path for a function called `formatUsingValidBscNativePath`.
One possibility would be to make the `formatUsingValidBscNativePath` function thinner, and avoid calling it at all unless there's a valid `bscNativePath`.

For example, instead of passing `isInterface` as a parameter, and creating the temp file internally, one could create the temp file externally, and pass it as a parameter.
Then move logic that calls the parser to the caller's site.

Reply:
What do you think about simply renaming it `formatCode`? That feels more in line with what it does now.

Reply:
Go for it!